pkg/config/options_test.go

```diff
@@ -266,68 +267,70 @@ func TestOperateOnCIOperatorConfigDir(t *testing.T) {
 	testCases := []struct {
 		id                     string
 		options                Options
-		expectedProcessedFiles sets.String
+		expectedProcessedFiles []string
 	}{
 		{
 			id:      "no options, expect to process all files",
 			options: Options{},
-			expectedProcessedFiles: sets.NewString([]string{
+			expectedProcessedFiles: []string{
 				"foo-bar-master.yaml",
 				"foo-bar-release-4.9.yaml",
 				"super-duper-master.yaml",
 				"super-duper-release-4.9.yaml",
-			}...),
+			},
 		},
 		{
 			id:      "specify org, expect to process only files that belong to that org",
 			options: Options{Org: "foo"},
-			expectedProcessedFiles: sets.NewString([]string{
+			expectedProcessedFiles: []string{
 				"foo-bar-master.yaml",
 				"foo-bar-release-4.9.yaml",
-			}...),
+			},
 		},
 		{
 			id:      "specify org and repo, expect to process only files that belong to that org/repo",
 			options: Options{Org: "foo", Repo: "bar"},
-			expectedProcessedFiles: sets.NewString([]string{
+			expectedProcessedFiles: []string{
 				"foo-bar-master.yaml",
 				"foo-bar-release-4.9.yaml",
-			}...),
+			},
 		},
 		{
 			id: "process only a single modified file",
 			options: Options{
 				onlyProcessChanges: true,
 				modifiedFiles:      sets.NewString([]string{"testdata/config/foo/bar/foo-bar-master.yaml"}...),
 			},
-			expectedProcessedFiles: sets.NewString([]string{
+			expectedProcessedFiles: []string{
 				"foo-bar-master.yaml",
-			}...),
+			},
 		},
 		{
 			id: "process only the multiple modified files",
 			options: Options{
 				onlyProcessChanges: true,
 				modifiedFiles:      sets.NewString([]string{"testdata/config/foo/bar/foo-bar-master.yaml", "testdata/config/super/duper/super-duper-release-4.9.yaml"}...),
 			},
-			expectedProcessedFiles: sets.NewString([]string{
+			expectedProcessedFiles: []string{
 				"foo-bar-master.yaml",
 				"super-duper-release-4.9.yaml",
-			}...),
+			},
 		},
 	}
 
 	for _, tc := range testCases {
 		t.Run(tc.id, func(t *testing.T) {
 			var errs []error
-			processed := sets.NewString()
-
+			var processed []string
 			if err := tc.options.OperateOnCIOperatorConfigDir(testConfigDir, func(configuration *cioperatorapi.ReleaseBuildConfiguration, info *Info) error {
 				filename := filepath.Base(info.Filename)
-				if !tc.expectedProcessedFiles.Has(filename) {
-					errs = append(errs, fmt.Errorf("file %s wasn't expected to be processed", filename))
+				for _, x := range tc.expectedProcessedFiles {
+					if x == filename {
+						processed = append(processed, filename)
+						return nil
+					}
 				}
-				processed.Insert(filename)
+				errs = append(errs, fmt.Errorf("file %s wasn't expected to be processed", filename))
```

Comment:
Hm, I see the dupe logic now - but why is this necessary? If you record all of the processed names, you should be able to see extras, missings and dupes in the cmp.Diff call

Reply:
Reimplemented in a less intrusive way, let me know if that does not address the comment.